networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_routing_driver.py

```diff
@@ -395,3 +457,76 @@ def _set_snat_pools_from_hosting_info(self, ri, gw_port, is_delete):
     def _set_nat_pool(self, ri, gw_port, is_delete):
         if gw_port['hosting_info'].get('snat_subnets'):
             self._set_snat_pools_from_hosting_info(ri, gw_port, is_delete)
+
+    def _handle_external_gateway_added_normal_router(self, ri, ext_gw_port):
```

Comment:
lots of duplicate code from parent's class, let's refactor and re-use as much code as possible then.